plugins/secrets-manager-aws-secret-manager/src/static/on_demand/providers/secrets/secretsManager.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { SecretsManagerServiceBase } from "./base/secretsManager.service.base";
+import { GetSecretValueCommand, SecretsManagerClient } from "@aws-sdk/client-secrets-manager";
+
+@Injectable()
+export class SecretsManagerService extends SecretsManagerServiceBase {
+  private readonly logger = new Logger()
+  private readonly awsSecretsClient: SecretsManagerClient
+
+  constructor(protected readonly configService: ConfigService) {
+    super(configService);
+
+    this.awsSecretsClient = new SecretsManagerClient({
+      region: configService.get("AWS_REGION")
+    })
+  }
+
+  async getSecret<T>(key: string): Promise<T | null> {
+    const [secret_id, secret_name] = key.split(":")
+
+    try {
+      const response = await this.awsSecretsClient.send(new GetSecretValueCommand({ SecretId: secret_id }))
+
+      const secrets_list: Record<string, string> = JSON.parse(
+        response.SecretString ? response.SecretString :
+        response.SecretBinary ? new TextDecoder().decode(response.SecretBinary) :
+        "{}"
+      )
+
+      return secrets_list[secret_name] as any
+    } catch(err: any) {
+      this.logger.warn(`Error while loading secret named "${secret_name}" from secret id "${secret_id}" - ${err.message}`, "SecretsManager")
+    }
```

Comment:
Be careful as we use ts. We will need to check for the error type otherwise we will get a build error like
<img width="1056" alt="image" src="https://github.com/amplication/plugins/assets/2861984/a89168f9-e7e3-494f-902d-17ce82813872">

I'd also suggest to log as ERROR instead of WARN as these might need immediate attention from the user